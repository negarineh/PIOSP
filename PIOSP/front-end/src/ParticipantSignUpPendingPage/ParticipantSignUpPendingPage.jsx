import React from 'react';
import '../components/stylesheets/MainPage1.css';
import '../components/stylesheets/MainPage.css';

import { Icon, Step} from 'semantic-ui-react';

// after page 2 when participants submit their name and email 
// they will redirect to this page as a pending page
class ParticipantSignUpPendingPage extends React.Component {
    
    render() {
      return (
            <div className="mainpage">
                <div className="wrappermainpage" > 
                      <div style={{textAlign:'center', margin:'40px'}}>
                        <h2 className='pheader'>Just one more step<br/></h2>
                          <p>Prove that you are not a robot to get your access to survey<br/></p>
                      </div>
                      <div style={{textAlign:'center', margin:'40px'}}>
                                <Step.Group size='large'>
                                  <Step completed>
                                  <Icon name='mail outline'/>
                                  <Step.Title>Step 1</Step.Title>
                                    <Step.Content>
                                      <p>Compeletd</p>
                                    </Step.Content>
                                  </Step>

                                  <Step active>
                                    <Icon name='mail outline' />
                                    <Step.Title style={{marginTop:'20px'}}>Step 2</Step.Title>
                                    <Step.Content>
                                      <p><br/>Go to your </p>
                                      <p>inbox and click  </p>
                                      <p>the confirmation </p> 
                                      <p>email we just</p>
                                      <p>sent you</p>
                                    </Step.Content>
                                  </Step>
                                
                                  <Step>
                                    <Icon name='file text' style={{color:'black'}}/>
                                    <Step.Title>Step 3</Step.Title>
                                    <Step.Content>
                                      <p>You'll </p>
                                      <p>get access </p>
                                      <p>to survey </p>
                                    </Step.Content>
                                  </Step>
                                </Step.Group>
                      </div>
            </div>
          </div>
      );
    }
  }


export  {ParticipantSignUpPendingPage};