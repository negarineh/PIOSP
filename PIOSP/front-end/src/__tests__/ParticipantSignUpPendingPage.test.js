import React from "react";
import Enzyme, { shallow } from "enzyme";
import assert from "assert";
import {ParticipantSignUpPendingPage} from "../ParticipantSignUpPendingPage";
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
 
Enzyme.configure({ adapter: new Adapter() });

// unit tests for the UserDeletePrompt component
describe('Participant SignUp Pending Page component', () => {
  describe('render()', () => {
    it('should render the component', () => {
      const wrapper = shallow(<ParticipantSignUpPendingPage/>);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      assert.equal(wrapper.length, 1);
    });
  });
});
