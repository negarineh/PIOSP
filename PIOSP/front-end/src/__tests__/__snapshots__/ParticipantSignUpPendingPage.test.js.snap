// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant SignUp Pending Page component render() should render the component 1`] = `
<div
  className="mainpage"
>
  <div
    className="wrappermainpage"
  >
    <div
      style={
        Object {
          "margin": "40px",
          "textAlign": "center",
        }
      }
    >
      <h2
        className="pheader"
      >
        Just one more step
        <br />
      </h2>
      <p>
        Prove that you are not a robot to get your access to survey
        <br />
      </p>
    </div>
    <div
      style={
        Object {
          "margin": "40px",
          "textAlign": "center",
        }
      }
    >
      <StepGroup
        size="large"
      >
        <Step
          completed={true}
        >
          <Icon
            as="i"
            name="mail outline"
          />
          <StepTitle>
            Step 1
          </StepTitle>
          <StepContent>
            <p>
              Compeletd
            </p>
          </StepContent>
        </Step>
        <Step
          active={true}
        >
          <Icon
            as="i"
            name="mail outline"
          />
          <StepTitle
            style={
              Object {
                "marginTop": "20px",
              }
            }
          >
            Step 2
          </StepTitle>
          <StepContent>
            <p>
              <br />
              Go to your 
            </p>
            <p>
              inbox and click  
            </p>
            <p>
              the confirmation 
            </p>
            <p>
              email we just
            </p>
            <p>
              sent you
            </p>
          </StepContent>
        </Step>
        <Step>
          <Icon
            as="i"
            name="file text"
            style={
              Object {
                "color": "black",
              }
            }
          />
          <StepTitle>
            Step 3
          </StepTitle>
          <StepContent>
            <p>
              You'll 
            </p>
            <p>
              get access 
            </p>
            <p>
              to survey 
            </p>
          </StepContent>
        </Step>
      </StepGroup>
    </div>
  </div>
</div>
`;
